WordPress/gutenberg
# [RNMobile] Fix crash when using the delete key to remove a single button

## What?
Fixes a crash when using the delete key to remove a single button:

* https://github.com/wordpress-mobile/gutenberg-mobile/issues/5388
* https://github.com/wordpress-mobile/gutenberg-mobile/pull/5855

## Why?
After typing text into a button and using the delete key to remove the text, it should also be possible to remove the button itself using the delete key.  Currently, using the delete key to remove a button results in a fatal exception. This occurs when the `onDeleteBlock` is sent as `undefined` to this code block:

https://github.com/WordPress/gutenberg/blob/c402cde955e9e280cb1f7fe8a8545eaaddb75abe/packages/block-library/src/button/edit.native.js#L327-L335

## How?
Adds missing `onDeleteBlock` prop to `block.native.js`. Adds test to check the button removal with the delete key (in addition to the existing test to use the "Remove Block" menu option).

## Testing Instructions
1. In the mobile app, create a post and add a single button.
2. Type test in the button, and then use the delete (backspace) key to remove the text and the button.
3. Observe that the button is removed, and the app does not crash.

## Screenshots or screencast 
Before | After
-|-
<video src="https://github.com/WordPress/gutenberg/assets/643285/ea0e562a-4ebe-4b1c-9533-a80198944b6f" /> | <video src="https://github.com/WordPress/gutenberg/assets/643285/9de40507-a557-4b05-a568-85b8567fb53c" />
